csharp/Adapter/Microsoft.Spark.CSharp/Network/RioNative.cs

```diff
@@ -128,25 +129,34 @@ private void Init()
             var maxThreads = 1;
             if (useThreadPool)
             {
-                maxThreads = Environment.ProcessorCount;
+                maxThreads = (Environment.ProcessorCount + 1) >> 1;
```

Comment:
As your mentioned, we can get the executor cores in cluster/standalone mode.

CSharpWorker [6260] is started with cores: 28